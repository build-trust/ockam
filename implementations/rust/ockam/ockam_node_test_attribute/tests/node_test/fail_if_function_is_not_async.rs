// This test checks that #[ockam_node_test_attribute::node_test] causes a compile time error
// if the item it is defined on is not an async function.

#[ockam_node_test_attribute::node_test]
fn my_test(c: ockam::Context) {
    c.stop().await.unwrap();
}
