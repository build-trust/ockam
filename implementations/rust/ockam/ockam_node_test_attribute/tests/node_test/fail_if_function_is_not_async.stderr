error: the `async` keyword is missing from the function declaration
 --> $DIR/fail_if_function_is_not_async.rs:5:1
  |
5 | fn my_test(c: ockam::Context) -> ockam::Result<()> {
  | ^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> $DIR/fail_if_function_is_not_async.rs:4:1
  |
4 | / #[ockam_node_test_attribute::node_test]
5 | | fn my_test(c: ockam::Context) -> ockam::Result<()> {
6 | |     c.stop().await.unwrap();
7 | | }
  | |_^ consider adding a `main` function to `$DIR/tests/node_test/fail_if_function_is_not_async.rs`
